src/main/java/com/wuest/prefab/structures/render/StructureRenderHandler.java

```diff
@@ -134,75 +143,75 @@ public static void renderPlayerLook(PlayerEntity player, HitResult src, MatrixSt
         }
     }
 
-    private static boolean renderComponentInWorld(World world, BuildBlock buildBlock, VertexConsumerProvider entityVertexConsumer, MatrixStack matrixStack, BlockPos pos) {
-        // Don't render this block if it's going to overlay a non-air/water block.
-        BlockState targetBlock = world.getBlockState(pos);
-        if (targetBlock.getMaterial() != Material.AIR && targetBlock.getMaterial() != Material.WATER) {
-            return false;
-        }
-
-        StructureRenderHandler.doRenderComponent(world, buildBlock, pos, entityVertexConsumer, matrixStack);
-
-        if (buildBlock.getSubBlock() != null) {
-            Block foundBlock = Registry.BLOCK.get(buildBlock.getSubBlock().getResourceLocation());
-            BlockState blockState = foundBlock.getDefaultState();
-
-            BuildBlock subBlock = BuildBlock.SetBlockState(
-                    StructureRenderHandler.currentConfiguration,
-                    world, StructureRenderHandler.currentConfiguration.pos,
-                    assumedNorth,
-                    buildBlock.getSubBlock(),
-                    foundBlock,
-                    blockState,
-                    StructureRenderHandler.currentStructure);
+    public static void render(MatrixStack stack) {
+        if (rendering) {
+            MinecraftClient minecraft = MinecraftClient.getInstance();
+            Camera camera = minecraft.getEntityRenderDispatcher().camera;
+            BlockPos pos = currentConfiguration.pos;
 
-            BlockPos subBlockPos = subBlock.getStartingPosition().getRelativePosition(
-                    StructureRenderHandler.currentConfiguration.pos,
-                    StructureRenderHandler.currentStructure.getClearSpace().getShape().getDirection(),
-                    StructureRenderHandler.currentConfiguration.houseFacing);
-
-            return StructureRenderHandler.renderComponentInWorld(world, subBlock, entityVertexConsumer, matrixStack, subBlockPos);
+            renderBlocks(vertexBuffer, pos, camera.getPos(), stack);
```

Comment:
Static function calls should be pre-pended with the class name to make it clearer who owns the static function.